org.adempiere.base/src/org/compiere/model/MDashboardContent.java

```diff
@@ -195,6 +216,17 @@ public static Map<String, String> parseProcessParameters(String parameters)	{
 				paramMap.put(key, value);
 			}
     	}
+    	
+    	// insert the multiselection values back to the HashMap
+    	if(multiSelections.size() > 0) {
+    		idx = 0;
+			for(Entry<String, String> e : paramMap.entrySet()) {
+				if(e.getValue().contains(placeHolder)) {
+					e.setValue(multiSelections.get(e.getValue()));
+					idx++;
+				}
+			}
+    	}

```

Comment:
something for consideration, if you keep a reverse map of paramMap.value to paramMap.key, say rmap, you can do a somewhat more efficient approach below:
`for(int i =0; i< idx; i++) {
   paramMap.put(rmap.getValue(placeHolder+i), multiSelections.get(placeHolder+i));
}`

Reply:
As I reviewed we don't even need a 3rd for loop, probably it would be even more efficient if we don't need to define the reversed map (see changes in the latest commit)